pkg/network/tcp_peer.go

```diff
@@ -293,7 +303,15 @@ func (p *TCPPeer) HandleVersion(version *payload.Version) error {
 		return errors.New("invalid handshake: already received Version")
 	}
 	p.version = version
-	p.lastBlockIndex = version.StartHeight
+	for _, cap := range version.Capabilities {
+		switch cap.Data.(type) {
```

Comment:
Why not `cap.Type == ` like it is done in the rest of the PR ?

Reply:
Fixed this.